src/View/Sidebar.vala

```diff
@@ -441,6 +441,46 @@ namespace Marlin.Places {
             return iter;
         }
 
+        public override Gtk.TreeIter add_or_update_plugin_item (PluginItem item, Gtk.TreeIter? iter = null) {
+            Gtk.TreeIter parent;
+            store.get_iter (out parent, network_category_reference.get_path ());
+
+            if (iter == null) {
+                store.append (out iter, parent);
+            } else if (!store.iter_is_valid (iter)) {
+                return iter;
+            }
+
```

Comment:
If this is only supposed to be used by plugin items, shouldn't we check that iter points to a plugin item?